.github/workflows/native-posix-tests.yml

```diff
@@ -0,0 +1,53 @@
+name: Native Posix tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+      - 'v*-branch'
+  push:
+    branches:
+      - main
+      - 'v*-branch'
+    tags:
+      - v*
+
+permissions:
+  contents: write
+  pull-requests: write
```

Comment:
why is write permission needed?

Reply:
> Since the workflow is not yet in the repo, it will not have permissions to post comment or upload the code coverage artifact. To see how this will work, refer https://github.com/balaji-nordic/sdk-nrf/pull/5

This will not work for the pull requests coming from forks even if this workflow is merged and in the repo.

If you want this to work for the fork pull requests, you need to use `pull_request_target` or isolate this step to a separate workflow triggering on `workflow_run`.

Reply:
> why is write permission needed?

I think this was left over code from the times when I was trying to figure out why artifacts wont get uploaded or the PR comment does not get published. I will remove this.

Reply:
> Do you suggest another workflow thats starts when this workflow completes? And that will be able to upload artifacts even if I dont use `pull_request_target`?

Yes, that is basically it. This workflow will run in the `pull_request` context which is limited in permissions, and another workflow will start after it completes via `workflow_run` event and this workflow will be running with write permissions required for commenting on pull requests.

You can see an example here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-build.yml
https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-publish-pr.yml

Also FYI, just uploading artifacts to a workflow run does not require "write permission."

Reply:
excellent. I will try that. Thanks

> Also FYI, just uploading artifacts to a workflow run does not require "write permission."

Got it.

Reply:
>I actually tried pull_request_target and github's behavior was absurd when I did that (The workflow run just disappeared from the PR without any warnings/errors). I will give it a shot again.

I guess it might be related with how the pull_request_target works. If it is used the workflow will use the code from the target, i.e. main in most cases. So the change has to be merged first and the result will be seen only on a following PRs. Maybe since there was no such workflow in the main, the action was not started on the PR introducing it. For pull_request (without _target) one can update the workflow within the PR and see the changes already in the same PR

Reply:
> Yes, that is basically it. This workflow will run in the `pull_request` context which is limited in permissions, and another workflow will start after it completes via `workflow_run` event and that workflow will be running with write permissions required for commenting on pull requests.
> 
> You can see an example here: https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-build.yml https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-publish-pr.yml

@stephanosio  I tried this. It does not work because apparently the workflow that should be triggered should exist on main branch and also it only runs on the code that is on main branch (this does not suit us in our case because we want the code coverage report to be done on the branch).

See : https://github.com/orgs/community/discussions/26062#discussioncomment-3250234

Reply:
> @stephanosio I tried this. It does not work because apparently the workflow that should be triggered should exist on main branch and also it only runs on the code that is on main branch (this does not suit us in our case because we want the code coverage report to be done on the branch).

If that is the case, `pull_request_target`, although it can be a security risk if not done right, would be the only way to do this. And, as @PerMac noted, it requires the pull request containing the workflow change to be merged first for it to work (you might need to test this in a separate "test" repository).

Reply:
I am concerned about using the `pull_request_target` too due to the security issues. github advises us to not even check out the source code in a workflow that uses `pull_request_target`. 

> Although the workflow runs in the context of the base of the pull request, you should make sure that you do not check out, build, or run untrusted code from the pull request with this event.

Source: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

Reply:
So perhaps writing a comment the the PR is unachievable as of now. Perhaps then I can remove that feature alone? So the work flow will only run native_posix tests? 

Side note: @thst-nordic  just told me that that github actions is way too slow in running anything because it only gives us 2 cpu cores. And we may be better off running it on jenkins that way we get faster feedback. Only disadvantage is that that they way we run unit tests (and native_posix tests) will become hidden from public.

Reply:
Using `pull_request_target` in itself should not be a problem as long as you are careful -- for example, make sure to not run any scripts from the repository checked out at the PR ref; if you have to (and you do because you need to run `twister`), check out a separate copy at the base ref.